impeller/entity/contents/clip_contents.cc

```diff
@@ -82,6 +82,20 @@ bool ClipContents::CanInheritOpacity(const Entity& entity) const {
 
 void ClipContents::SetInheritedOpacity(Scalar opacity) {}
 
+bool ClipContents::CanSkip(ISize render_pass_size, const Matrix& ctm) const {
+  if (clip_op_ == Entity::ClipOperation::kIntersect &&
+      geometry_->IsAxisAlignedRect() && ctm.IsTranslationScaleOnly()) {
+    std::optional<Rect> coverage = geometry_->GetCoverage(ctm);
+    if (coverage.has_value() &&
+        coverage->Contains(Rect::MakeSize(render_pass_size))) {
+      // Skip axis-aligned intersect clips that cover the whole render target
+      // since they won't draw anything to the depth buffer.
```

Comment:
The DL code can deal with non-axis alignment but unfortunately it doesn't have access to this information. If we are axis aligned then this is the fast answer, but if we aren't then a slightly slower answer is to reverse transform the existin clip bounds *corners* by the ctm and then see if they are inside the geometry.

See: https://github.com/flutter/engine/blob/09fa10157ca839e27e732f943f0b63b4fdc26be0/display_list/utils/dl_matrix_clip_tracker.cc#L253

Also, it uses a different test - IsAligned2D instead of IsTranslationScaleOnly. Aligned allows for 90 degree rotations I think.

Reply:
Also, why only do this on the render pass size? Why not skip it if it covers the entire existing clip bounds?

Reply:
hmm, we actually are also checking the clip bounds. These should already include the render pass size.

https://github.com/flutter/engine/blob/main/impeller/aiks/experimental_canvas.cc#L769-L797

Maybe this logic isn't necessary at all

Reply:
Ahh, so the problem is that we _always_ report the clip coverage as changing:

https://github.com/flutter/engine/blob/main/impeller/entity/entity_pass_clip_stack.cc#L69-L82

Even if the new clip coverage is bigger than the old clip coverage. We could move this check into the clip coverage stack and simplify this everywhere.

Reply:
Yeah, lets just move this logic into the clip coverage stack